src/components/@shared/DDODownloadButton/inputWithList.tsx

```diff
@@ -0,0 +1,79 @@
+import { getFieldContent } from '@utils/form'
+import content from '../../../../content/asset/form.json'
+import { Field, useFormikContext } from 'formik'
+import Input from '../FormInput'
+import Button from '../atoms/Button'
+import { useState } from 'react'
+import styles from './index.module.css'
+import { ListItem } from '../atoms/Lists'
+
+export default function InputWithList({
+  fieldname,
+  setList,
+  list
+}: {
+  fieldname: string
+  setList: (list: { id: string }[]) => void
+  list: { id: string }[]
+}) {
+  const [serviceCredential, setServiceCredential] = useState<string>()
+  const { setFieldValue, errors } = useFormikContext()
+  // add credential to the credentiallist
+  const addCredential = () => {
+    if (serviceCredential.trim() && !errors[fieldname]) {
+      setList([...list, { id: serviceCredential }])
+      setServiceCredential('')
+    }
+  }
+  // delete credential from the list
+  const deleteCredential = (index: number) => {
+    list.splice(index, 1)
```

Comment:
nitpick (blocking): The use of splice directly mutates the list array. Mutating state directly can cause issues with React’s rendering logic. Use filter instead of splice to avoid mutating the state directly.